awscli/customizations/commands.py

```diff
@@ -177,7 +177,8 @@ def __call__(self, args, parsed_globals):
                     cli_argument.argument_model, value)
 
             setattr(parsed_args, key, value)
-
+        if hasattr(self._session, 'user_agent_extra'):
```

Comment:
I'm wonder if we should add this to the auto-generated commands too. One side effect of only doing it in this class is that you can get in weird cases like with the `wait` commands:
```
$ aws s3api wait bucket-exists --bucket foo --debug
```
where because the `wait` component is subclassed from a custom command class but the `bucket-exists` class subclasses from the `ServiceOperation` class, it misses the `bucket-exists` command in the user agent: `command/s3api.wait`